lib/pure/asyncmacro.nim

```diff
@@ -440,7 +399,26 @@ macro multisync*(prc: untyped): untyped =
   result.add(asyncSingleProc(asyncPrc))
   result.add(sync)
 
-proc await*[T](x: T) =
-  ## The 'await' keyword is also defined here for technical
-  ## reasons. (Generic symbol lookup prepass.)
-  {.error: "Await only available within .async".}
+# seems simpler to define it here, than to define it in each multisync
+template await*(f: untyped): untyped =
+  when declared(internalRetFuture):
+    error "Await expects future"
+  elif not declared(inMultisync):
+    error "Await only available within {.async.}"
+  else:
+    f
+ 
+# based on the yglukhov's patch to chronos: https://github.com/status-im/nim-chronos/pull/47
+template await*[T](f: Future[T]): auto =
+  when declared(internalRetFuture):
+    when not declared(internalTmpFuture):
+      var internalTmpFuture: FutureBase
+    internalTmpFuture = f
+    # internalRetFuture.child = internalTmpFuture
+    yield internalTmpFuture
+    if not isNil(internalTmpFuture.error):
+      echo "error"
```

Comment:
Remove the `echo` :grey_question:

Reply:
sounds reasonable, i don't remember why i had it